tests/Kernel/CorporateBlocks/CorporateBlocksFooterTest.php

```diff
@@ -5,316 +5,182 @@
+    $actual = $section->filter('.ecl-footer-core__title');
+    $this->assertEquals('Legal links title', $actual->text());
 
-    // Make sure that footer block is present.
-    $actual = $crawler->filter('footer.ecl-footer');
-    $this->assertCount(1, $actual);
+    $this->assertLinkList($section, 'core', $test_data['legal_links']);
 
-    // Make sure that footer block rendered correctly.
-    $actual = $crawler->filter('footer.ecl-footer div.ecl-footer__sections div.ecl-row section.ecl-footer__section');
-    $this->assertCount(2, $actual);
+    $section = $crawler->filter('footer.ecl-footer-core section.ecl-footer-core__section4');
 
-    $first_column = $crawler->filter('footer.ecl-footer div.ecl-footer__sections div.ecl-row section.ecl-footer__section:nth-child(1)');
+    $actual = $section->filter('.ecl-footer-core__title');
+    $this->assertEquals('Institution links title', $actual->text());
 
-    $actual = $first_column->filter('h1.ecl-footer__section-title');
-    $this->assertEquals('Contact title', trim($actual->first()->text()));
-
-    $actual = $first_column->filter('ul.ecl-footer__section-list li.ecl-footer__section-item:nth-child(1)');
-    $this->assertEquals('Contact 1 <a class="ecl-footer__section-link ecl-link ecl-link--standalone" href="#">link</a>', trim($actual->html()));
-
-    $actual = $first_column->filter('ul.ecl-footer__section-list li.ecl-footer__section-item:nth-child(2)');
-    $this->assertEquals('Contact 2 <a class="ecl-footer__section-link ecl-link ecl-link--standalone" href="#">link</a>', trim($actual->html()));
-
-    $actual = $first_column->filter('h1.ecl-footer__section-title');
-    $this->assertEquals('Social media title', trim($actual->last()->text()));
-
-    $second_column = $crawler->filter('footer.ecl-footer div.ecl-footer__sections div.ecl-row section.ecl-footer__section:nth-child(2)');
-    $actual = $second_column->filter('h1.ecl-footer__section-title');
-    $this->assertEquals('Institution links title', trim($actual->text()));
-
-    $actual = $second_column->filter('ul.ecl-footer__section-list li.ecl-footer__section-item:nth-child(1) > a');
-    $this->assertEquals('https://europa.eu/institution_links1', $actual->attr('href'));
-    $this->assertEquals('Institution link 1', trim($actual->text()));
+    $this->assertLinkList($section, 'core', $test_data['institution_links']);
+  }
 
-    $actual = $second_column->filter('ul.ecl-footer__section-list li.ecl-footer__section-item:nth-child(2) > a');
-    $this->assertEquals('https://europa.eu/institution_links2', $actual->attr('href'));
-    $this->assertEquals('Institution link 2', trim($actual->text()));
+  /**
+   * Test European Union footer standardised block rendering.
+   */
+  public function testEuFooterStandardisedBlockRendering(): void {
+    $this->configFactory->getEditable('oe_theme.settings')->set('branding', 'standardised')->save();
+    $test_data = [];
+    $html = $this->renderCorporateBlocksFooter('eu', $test_data);
```

Comment:
This method render the block and fetches the test data, isn't it more clear for one method to do one thing only? Let's simply:

```php
    $test_data = $this->getFixtureContent("{$type}_footer.yml");
    $html = $this->renderCorporateBlocksFooter($type, $test_data);
```